src/utils/client.js

```diff
@@ -164,7 +164,7 @@ function getMismatchedClientFields(echisClientDoc, crClientDoc) {
   matcherFields.forEach(key => {
     const actual = getPropByString(crPayload, key);
     const expected = getPropByString(crClientDoc, key);
-    if (actual.toUpperCase() !== expected.toUpperCase()) {
+    if (actual?.toUpperCase() !== expected?.toUpperCase()) {
```

Comment:
This just hides the problem. Can you confirm what field was missing? Is it related to https://github.com/moh-kenya/echis-2.0-integration/issues/126?

Reply:
I get that but in the error you shared it was failing because `expected` was undefined, correct? `expected` is a field from the CR client doc so that means `getPropByString` is returning undefined.

Reply:
Reverted the changes.

Reply:
The mediator has updated the missing fields on eCHIS as shown below and works in the expected behavior. The problem is the service dies and returns a 502 Bad Gateway. 

![image](https://github.com/moh-kenya/echis-2.0-integration/assets/18611345/264872cc-2538-4c92-b695-6a5967d2185d)

Reply:
> The problem is the service dies and returns a 502 Bad Gateway.

This is what we're trying to resolve, it will crash again if we don't know what caused the initial error. Can you link me one example transaction i can use to replicate the failure you were getting in the demo?

Reply:
Resolved it under this review:

https://github.com/moh-kenya/echis-2.0-integration/pull/125/#discussion_r1356916315